File: BOJ/9935/A.cpp
#include <iostream>
#include <deque>
using namespace std;

int main()
{
    freopen("input.txt", "rt", stdin);
    ios::sync_with_stdio(0);
    cin.tie(0);
    string a, b;
    cin >> a >> b;
    string result;
    for (char ch : a)
    {
        result.push_back(ch);
        if (result.size() >= b.size() && result.substr(result.size() - b.size()) == b)
        {
            result.erase(result.size() - b.size());
        }
    }
    if (result.empty())
    {
        cout << "FRULA";
    }
    else
    {
        cout << result;
    }
}

File: BOJ/9935/A.py
import sys
sys.stdin = open("D:\Projects\AA\BOJ\9935\input.txt")
input = sys.stdin.readline
S = str(input().rstrip())
bomb = str(input().rstrip())
st = S
l = len(bomb)
result = ""
i = 0
while i <= len(st)-l:
    test = st[i:i+l]
    j = i
    while test == bomb:
        st = st[0:j] + st[j+l:]
        j = i - l
        if j < 0:
            j = 0
        test = st[j:j+l]
    i = j
    i = i + 1

if st == "":
    print("FRULA")
else:
    print(st)
